src/flameGraph.cpp

```diff
@@ -24,239 +24,10 @@
-    "\t\t\t\t\tc.fillStyle = 'rgba(255, 255, 255, 0.5)';\n"
-    "\t\t\t\t\tc.fillRect((f.left - x0) * px, y, f.width * px, 15);\n"
-    "\t\t\t\t}\n"
-    "\t\t\t}\n"
-    "\t\t}\n"
-    "\n"
-    "\t\tfor (let h = 0; h < levels.length; h++) {\n"
-    "\t\t\tconst y = reverse ? h * 16 : canvasHeight - (h + 1) * 16;\n"
-    "\t\t\tconst frames = levels[h];\n"
-    "\t\t\tfor (let i = 0; i < frames.length; i++) {\n"
-    "\t\t\t\tdrawFrame(frames[i], y, h < rootLevel);\n"
-    "\t\t\t}\n"
-    "\t\t}\n"
-    "\n"
-    "\t\treturn totalMarked();\n"
-    "\t}\n"
-    "\n"
-    "\tcanvas.onmousemove = function() {\n"
-    "\t\tconst h = Math.floor((reverse ? event.offsetY : (canvasHeight - event.offsetY)) / 16);\n"
-    "\t\tif (h >= 0 && h < levels.length) {\n"
-    "\t\t\tconst f = findFrame(levels[h], event.offsetX / px + root.left);\n"
-    "\t\t\tif (f) {\n"
-    "\t\t\t\thl.style.left = (Math.max(f.left - root.left, 0) * px + canvas.offsetLeft) + 'px';\n"
-    "\t\t\t\thl.style.width = (Math.min(f.width, root.width) * px) + 'px';\n"
-    "\t\t\t\thl.style.top = ((reverse ? h * 16 : canvasHeight - (h + 1) * 16) + canvas.offsetTop) + 'px';\n"
-    "\t\t\t\thl.firstChild.textContent = f.title;\n"
-    "\t\t\t\thl.style.display = 'block';\n"
-    "\t\t\t\tcanvas.title = f.title + '\\n(' + samples(f.width) + f.details + ', ' + pct(f.width, levels[0][0].width) + '%%)';\n"
-    "\t\t\t\tcanvas.style.cursor = 'pointer';\n"
-    "\t\t\t\tcanvas.onclick = function() {\n"
-    "\t\t\t\t\tif (f != root) {\n"
-    "\t\t\t\t\t\trender(f, h);\n"
-    "\t\t\t\t\t\tcanvas.onmousemove();\n"
-    "\t\t\t\t\t}\n"
-    "\t\t\t\t};\n"
-    "\t\t\t\tstatus.textContent = 'Function: ' + canvas.title;\n"
-    "\t\t\t\treturn;\n"
-    "\t\t\t}\n"
-    "\t\t}\n"
-    "\t\tcanvas.onmouseout();\n"
-    "\t}\n"
-    "\n"
-    "\tcanvas.onmouseout = function() {\n"
-    "\t\thl.style.display = 'none';\n"
-    "\t\tstatus.textContent = '\\xa0';\n"
-    "\t\tcanvas.title = '';\n"
-    "\t\tcanvas.style.cursor = '';\n"
-    "\t\tcanvas.onclick = '';\n"
-    "\t}\n"
-    "\n"
-    "\tdocument.getElementById('reverse').onclick = function() {\n"
-    "\t\treverse = !reverse;\n"
-    "\t\trender();\n"
-    "\t}\n"
-    "\n"
-    "\tdocument.getElementById('search').onclick = function() {\n"
-    "\t\tsearch(true);\n"
-    "\t}\n"
-    "\n"
-    "\tdocument.getElementById('reset').onclick = function() {\n"
-    "\t\tsearch(false);\n"
-    "\t}\n"
-    "\n"
-    "\twindow.onkeydown = function() {\n"
-    "\t\tif (event.ctrlKey && event.keyCode === 70) {\n"
-    "\t\t\tevent.preventDefault();\n"
-    "\t\t\tsearch(true);\n"
-    "\t\t} else if (event.keyCode === 27) {\n"
-    "\t\t\tsearch(false);\n"
-    "\t\t}\n"
-    "\t}\n";
-
-static const char FLAMEGRAPH_FOOTER[] =
-    "render();\n"
-    "</script></body></html>\n";
-
+static const char FLAMEGRAPH_TEMPLATE[] = {
+#include "helper/one/profiler/flame.html.h"
+0
+};
 
 static const char TREE_HEADER[] =
```

Comment:
I suppose it makes sense to convert this to a template as well.

Reply:
Done